### DISCUSSION TITLE
Can I start the WebUI from the experiment folder?

### DISCUSSION THREAD

**Discussion by weiT1993** [microsoft/nni#3787] [Ideas]
Can I run an NNI experiment on a remote machine, download the experiment folder after it's done to local, and view the results in WebUI from local?

The remote machine I use requires me to do a second ssh from a remote server to log in. I cannot start a browser from the remote server.

**Comment by QuanluZhang** [comment]
@weiT1993 , thanks for reporting your requirement. Unfortunately, NNI has not supported this feature. We have supported [experiment view mode](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#view), in this mode an experiment can be view through webui without resuming the experiment. Your requirement is one more step further: first migrate the experiment data, second view the experiment through webui. We will consider supporting this feature.

**Comment by scarlett2018** [comment]
> @weiT1993 , thanks for reporting your requirement. Unfortunately, NNI has not supported this feature. We have supported [experiment view mode](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#view), in this mode an experiment can be view through webui without resuming the experiment. Your requirement is one more step further: first migrate the experiment data, second view the experiment through webui. We will consider supporting this feature.

is `remote `mode easier for his case?

**Comment by weiT1993** [comment]
> > @weiT1993 , thanks for reporting your requirement. Unfortunately, NNI has not supported this feature. We have supported [experiment view mode](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#view), in this mode an experiment can be view through webui without resuming the experiment. Your requirement is one more step further: first migrate the experiment data, second view the experiment through webui. We will consider supporting this feature.
> 
> is `remote `mode easier for his case?

I did not figure out how to use the remote mode properly. The remote server 2 I use quires me to first ```ssh``` into server 1 and ```ssh``` from there to log in.  I used ```ping``` to find the IP of server 2 but that does not work in the remote mode. Are there guidelines on how to start remote mode for servers that I can ```ssh``` into?

**Comment by scarlett2018** [comment]
> > > @weiT1993 , thanks for reporting your requirement. Unfortunately, NNI has not supported this feature. We have supported [experiment view mode](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#view), in this mode an experiment can be view through webui without resuming the experiment. Your requirement is one more step further: first migrate the experiment data, second view the experiment through webui. We will consider supporting this feature.
> > 
> > 
> > is `remote `mode easier for his case?
> 
> I did not figure out how to use the remote mode properly. The remote server 2 I use quires me to first `ssh` into server 1 and `ssh` from there to log in. I used `ping` to find the IP of server 2 but that does not work in the remote mode. Are there guidelines on how to start remote mode for servers that I can `ssh` into?

can your local machine open up web ui from server 1? if so, you can run remote mode with server 2 on server 1. and then access server 1's NNI web ui.

**Comment by weiT1993** [comment]
> > > > @weiT1993 , thanks for reporting your requirement. Unfortunately, NNI has not supported this feature. We have supported [experiment view mode](https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/Nnictl.md#view), in this mode an experiment can be view through webui without resuming the experiment. Your requirement is one more step further: first migrate the experiment data, second view the experiment through webui. We will consider supporting this feature.
> > > 
> > > 
> > > is `remote `mode easier for his case?
> > 
> > 
> > I did not figure out how to use the remote mode properly. The remote server 2 I use quires me to first `ssh` into server 1 and `ssh` from there to log in. I used `ping` to find the IP of server 2 but that does not work in the remote mode. Are there guidelines on how to start remote mode for servers that I can `ssh` into?
> 
> can your local machine open up web ui from server 1? if so, you can run remote mode with server 2 on server 1. and then access server 1's NNI web ui.

Unfortunately no. I cannot open up web ui from either server 1 or server 2.

**Comment by ragavvenkatesan** [comment]
I am interested in this as well. Seems like an obvious feature addition.

**Comment by QuanluZhang** [comment]
@ragavvenkatesan thanks for echoing this feature, we will consider this feature in future release. Also welcome external contributions.

**Comment by steermomo** [comment]
Is there any way to view trial results directly in the console? Such as display  trial's final result report via `nnctl top`

**Comment by scarlett2018** [comment]
> Is there any way to view trial results directly in the console? Such as display trial's final result report via `nnctl top`

does `nnictl trial ls` or `nnicli` (import nnicli) work for you?

**Comment by Lamply** [comment]
Really need this feature. It's too hard to use NNI in private training platform.

**Comment by QuanluZhang** [comment]
> Really need this feature. It's too hard to use NNI in private training platform.

@Lamply [`nnictl experiment save/load`](https://nni.readthedocs.io/en/stable/Tutorial/Nnictl.html#experiment) should meet your requirement. You can save the experiment executed in the private training platform, download the saved, and load it in your local machine. then use [`nnictl view`](https://nni.readthedocs.io/en/stable/Tutorial/Nnictl.html#view) to view your experiment in local.

**Reply by Lamply** [reply]
Thanks! Though have finished this using ray, I will check it out next time.

**Reply by QuanluZhang** [reply]
@Lamply , thanks for your reply. May I know how ray solves your problem that running trials on remote machine cannot be easily visualized? It would helps us to improve NNI :)

**Reply by Lamply** [reply]
Ray Tune support logging to TensorBoard. I just need to set `local_dir` in `ray.tune.run` to save training results to my cloud storage, and report any values need to be visualized using `ray.tune.report`. Then I can start training on remote machine, visualize all trials and corresponding metric curves in any machine that can access the cloud storage.

**Reply by QuanluZhang** [reply]
@Lamply , very helpful, thanks! we will rethink the design for this feature, and post the design here.

**Comment by achillesliu** [comment]
Now is there any updates about it?
